src/test/scala-2.11/com/linagora/gatling/imap/scenario/ImapFullScenario.scala

```diff
@@ -0,0 +1,67 @@
+package com.linagora.gatling.imap.scenario
+
+import java.util.Calendar
+
+import com.linagora.gatling.imap.PreDef._
+import com.linagora.gatling.imap.protocol.Uid
+import com.linagora.gatling.imap.protocol.command.FetchAttributes.AttributeList
+import com.linagora.gatling.imap.protocol.command.FetchRange._
+import io.gatling.app.Gatling
+import io.gatling.core.Predef._
+import io.gatling.core.config.GatlingPropertiesBuilder
+import io.gatling.core.scenario.Simulation
+
+import scala.collection.immutable.Seq
+import scala.concurrent.duration._
+
+class ImapFullScenario extends Simulation {
+  val feeder = csv("users.csv").circular
+  val userCount: Int = feeder.records.length
+
+  val receiveEmail = exec(imap("append").append("INBOX", Some(Seq("\\Flagged")), Option.empty[Calendar],
+                  """From: expeditor@example.com
+                    |To: recipient@example.com
+                    |Subject: test subject
+                    |
+                    |Test content
+                    |abcdefghijklmnopqrstuvwxyz
+                    |0123456789""".stripMargin).check(ok))
+
+  val readLastEmail = exec(imap("list").list("", "*").check(ok, hasFolder("INBOX")))
+                      .exec(imap("select").select("INBOX").check(ok))
+                      .exec(imap("fetch").fetch(Seq(Last()), AttributeList("BODY[HEADER]", "UID", "BODY[TEXT]")).check(ok))
+
+  val heavyUser = repeat(3)(receiveEmail)
+                  .repeat(2)(readLastEmail)
+  val lightUser = receiveEmail
+                  .exec(readLastEmail)
+
+  val duration = 4 hours
+
+  val scn = scenario("Imap").feed(feeder)
+    .pause(1 second)
+    .exec(imap("Connect").connect()).exitHereIfFailed
+    .exec(imap("login").login("${username}", "${password}").check(ok))
+    .exec {
+      randomSwitch(
+        75.0 -> during(duration) {
+                  pace(5 minutes, 15 minutes)
+                  .exec(lightUser)
+                },
+
+        25.0 -> during(duration) {
+                  pace(30 seconds, 90 seconds)
+                  .exec(heavyUser)
+                }
+      )
+
+    }
+
+  val userCountWith50PercentsOfThemAlsoReadingTheirEmailOnAnotherDevice = (userCount * 1.5).toInt
+
+  setUp(scn.
+      inject(
+          nothingFor(1 seconds),
+          rampUsers(userCountWith50PercentsOfThemAlsoReadingTheirEmailOnAnotherDevice) over(10 minutes)))
```

Comment:
Ok I will add a maxDuration. I didn't care of the 10 minutes + some seconds :)
For the throttle, for me it's an other test. In this test I want to check than some thousands of users can use IMAP during 4 hours, and have a look of the behaviour of the server in this test.
Your proposal is more about checking the server can sustain some specific load during some times.
I could write such a test later, but that's an other issue for me.